File: fasttext_repo/ft_main.py
import ft_config
from ft_train import train_fasttext_model

def main() -> None:
    """
    given nothing
    return nothing
    executes fasttext training pipeline
    """
    train_fasttext_model(ft_config.TRAIN_FILE, ft_config.MODEL_FILE)

if __name__ == "__main__":
    main()

File: src/models/fasttext/fasttext_repo/ft_main.cpp
#include <iostream>
#include "fasttext.h"
#include "ft_config.hpp"
#include "ft_predict.hpp"

/*
given nothing
return integer exit status
loads model and runs prediction
*/
int main() {
    fasttext::FastText model;
    model.loadModel(MODEL_FILE);
    predict_label(model, "this is a great product");
    return 0;
}
